src/lib/storage/vector_compression/simd_bp128/simd_bp128_iterator.cpp

```diff
@@ -2,76 +2,38 @@
 
 namespace opossum {
 
-SimdBp128Iterator::SimdBp128Iterator(const pmr_vector<uint128_t>* data, const size_t size, const size_t absolute_index)
-    : _data{data},
-      _size{size},
-      _data_index{absolute_index},
-      _absolute_index{absolute_index},
-      _current_meta_block{std::make_unique<std::array<uint32_t, Packing::meta_block_size>>()},
-      _current_meta_block_index{0u} {
-  if (data && !data->empty() && _absolute_index != _size) {
-    _unpack_next_meta_block();
-  }
-}
+SimdBp128Iterator::SimdBp128Iterator(SimdBp128Decompressor&& decompressor, const size_t absolute_index)
+    : _decompressor{std::move(decompressor)}, _absolute_index{absolute_index} {}
 
 SimdBp128Iterator::SimdBp128Iterator(const SimdBp128Iterator& other)
-    : _data{other._data},
-      _size{other._size},
-      _data_index{other._data_index},
-      _absolute_index{other._absolute_index},
-      _current_meta_block{std::make_unique<std::array<uint32_t, Packing::meta_block_size>>(*other._current_meta_block)},
-      _current_meta_block_index{other._current_meta_block_index} {}
+    : _decompressor{SimdBp128Decompressor(other._decompressor)}, _absolute_index{other._absolute_index} {}
+
+SimdBp128Iterator::SimdBp128Iterator(SimdBp128Iterator&& other) noexcept
+    : _decompressor{std::move(other._decompressor)}, _absolute_index{other._absolute_index} {}
 
 SimdBp128Iterator& SimdBp128Iterator::operator=(const SimdBp128Iterator& other) {
   if (this == &other) return *this;
-  _data = other._data;
-  _size = other._size;
-  _data_index = other._data_index;
+
+  _decompressor = SimdBp128Decompressor(other._decompressor);
   _absolute_index = other._absolute_index;
-  _current_meta_block = std::make_unique<std::array<uint32_t, Packing::meta_block_size>>(*other._current_meta_block);
-  _current_meta_block_index = other._current_meta_block_index;
   return *this;
 }
 
-void SimdBp128Iterator::_unpack_next_meta_block() {
-  _read_next_meta_info();
-
-  for (uint8_t meta_info_index = 0u; meta_info_index < Packing::blocks_in_meta_block; ++meta_info_index) {
-    _unpack_block(meta_info_index);
-  }
-
-  _current_meta_block_index = 0u;
-}
+// As we use boost's iterator facades, we cannot use _increment().
```

Comment:
Rather: “Our code style would want this to be _increment, but we need to implement boost’s interface. Same for the methods below.”?